Build fails with clang on GLM_DEPRECATED GLM_FUNC_DECL construct

I've ran into this when build-testing one of glm consumers, gamescope, before updating FreeBSD port of glm to 1.x.

```
c++ -Isrc/gamescope_color_tests.p -Isrc -I../src -I/usr/local/include -fdiagnostics-color=never -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++20 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-missing-f
ield-initializers -Wno-c99-designator -Wno-invalid-offsetof -Wno-unused-const-variable -Wno-deprecated-volatile -Wno-ignored-qualifiers -Wno-missing-braces -ffast-math -DHAVE_PIPEWIRE=1 -DHAVE_OPENVR=0 '-DHWDATA_P
NP_IDS="//usr/local/share/hwdata/hwdata/pnp.ids"' -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DGLM_ENABLE_EXPERIMENTAL -MD -MQ src/gamescope_color_tests.p/color_tests.cpp.o -MF src/gamescope_color_tes
ts.p/color_tests.cpp.o.d -o src/gamescope_color_tests.p/color_tests.cpp.o -c ../src/color_tests.cpp
In file included from ../src/color_tests.cpp:4:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:49:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL genIUType powerOfTwoAbove(genIUType Value);
                       ^
/usr/local/include/glm/./ext/../detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/./ext/../detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
```

Reproducible with
```c++
#define GLM_ENABLE_EXPERIMENTAL
#include <glm/gtx/bit.hpp>
```

What preprocessed line looks like:
```c++
__attribute__((__deprecated__)) [[nodiscard]] genIUType powerOfTwoAbove(genIUType Value);
```

Smaller repro (note that template is required):
```c++
template<class T> __attribute__((__deprecated__)) [[nodiscard]] int foo(T t);
```
```
% clang++-14 -std=c++20  -c 1.cc 
1.cc:1:51: error: an attribute list cannot appear here
template<class T> __attribute__((__deprecated__)) [[nodiscard]] int foo(T t);
                                                  ^~~~~~~~~~~~~
1 error generated.
```

I haven't dug into the docs, but simple experiment revealed two ways to fix it: either change the order of attributes or use attributes of the same format, e.g. change `__attribute__((__deprecated__))` to `[[deprecated]`. Both compile by gcc and clang, and I suggest to use `[[deprecated]]` in glm, as it's standard and supported since `c++14`.

My builds are failing too, I think it's because Homebrew has moved to verison 1.0.1

https://github.com/pemsley/coot/actions/workflows/build-coot-macos.yml

```In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:49:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL genIUType powerOfTwoAbove(genIUType Value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:57:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL vec<L, T, Q> powerOfTwoAbove(vec<L, T, Q> const& value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:65:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL genIUType powerOfTwoBelow(genIUType Value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:73:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL vec<L, T, Q> powerOfTwoBelow(vec<L, T, Q> const& value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:81:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL genIUType powerOfTwoNearest(genIUType Value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.cpp:10:
In file included from ../../../coot/MoleculesToTriangles/CXXClasses/RepresentationInstance.h:15:
In file included from /usr/local/include/glm/ext.hpp:198:
/usr/local/include/glm/./gtx/bit.hpp:89:17: error: 'nodiscard' attribute cannot be applied to types
        GLM_DEPRECATED GLM_FUNC_DECL vec<L, T, Q> powerOfTwoNearest(vec<L, T, Q> const& value);
                       ^
/usr/local/include/glm/detail/setup.hpp:498:23: note: expanded from macro 'GLM_FUNC_DECL'
#define GLM_FUNC_DECL GLM_NODISCARD GLM_CUDA_FUNC_DECL
                      ^
/usr/local/include/glm/detail/setup.hpp:344:26: note: expanded from macro 'GLM_NODISCARD'
#       define GLM_NODISCARD [[nodiscard]]
                               ^
6 errors generated.
make[2]: *** [libMoleculesToTrianglesCXXClasses_la-RepresentationInstance.lo] Error 1
```

--- a/glm/detail/setup.hpp
+++ b/glm/detail/setup.hpp
@@ -594,7 +594,11 @@
 #	define GLM_DEPRECATED __declspec(deprecated)
 #	define GLM_ALIGNED_TYPEDEF(type, name, alignment) typedef __declspec(align(alignment)) type name
 #elif GLM_COMPILER & (GLM_COMPILER_GCC | GLM_COMPILER_CLANG | GLM_COMPILER_INTEL)
-#	define GLM_DEPRECATED __attribute__((__deprecated__))
+#	if GLM_LANG & GLM_LANG_CXX14_FLAG
+#		define GLM_DEPRECATED [[deprecated]]
+#	else
+#		define GLM_DEPRECATED __attribute__((__deprecated__))
+#	endif
 #	define GLM_ALIGNED_TYPEDEF(type, name, alignment) typedef type name __attribute__((aligned(alignment)))
 #elif (GLM_COMPILER & GLM_COMPILER_CUDA) || (GLM_COMPILER & GLM_COMPILER_HIP)
 #	define GLM_DEPRECATED

